downloader/src/main/kotlin/Downloader.kt

```diff
@@ -80,7 +83,35 @@ object Downloader {
 
         val exception = DownloadException("Download failed for '${pkg.id.toCoordinates()}'.")
 
-        // Try downloading from VCS.
+        config.sourceCodeOrigins.forEach { origin ->
```

Comment:
I'd re-introduce `distinct()` here to avoid the need for the previously introduced check.

Reply:
It was removed due to @mnonnenmacher's comment: https://github.com/oss-review-toolkit/ort/pull/3736#discussion_r599442793
And I agree with this approach:
> then other places do not have to verify that it does not contain duplicates

Reply:
Sorry, I've overlooked that. I tend to disagree, as that list is only supposed to configure the relative order (and mentioning the same source multiple times does not change the order), so I believe we could be lenient about such a configuration glitch for user convenience. But I'm fine to be outvoted here.

Reply:
Configuring the following should be detected and prevented because we cannot know the user's intended order:
```
downloader {
  sourceCodeOrigins: [
    "VCS", "ARTIFACT", "VCS"
  ]
}
```
De-duplicating at a later point would not catch this configuration error.